src/test/kotlin/app/services/impl/HttpKeyServiceTest.kt

```diff
@@ -53,7 +52,7 @@ class HttpKeyServiceTest {
     }
 
     @Test
-    fun testDataKeyOk() {
+    fun testBatchDataKey_WillCallClient_AndReturnKey() {
```

Comment:
Do we also need tests to check if the delay between retrys is actually 1000ms?

Reply:
Imho, no, as that is testing Sprint it's self, which is reliable.